README.md

```diff
@@ -326,6 +326,59 @@ if err != nil {
 fmt.Print(status, status.ChannelId)
 ```
 
+### Authentication
```

Comment:
Other than the code snippets, this documentation does not seem to be specific to ably-go, so I don't think we should be explaining it here in the README, I think we should be linking to sections of our official docs (and if there are sections missing from our docs, let's add to those docs rather than fill the gap in a single SDK's README).

For example, what is said below:

> `ABLY_KEY` should not be exposed at client side where it can be used for malicious purposes.

is explained here:

https://ably.com/docs/auth#mechanisms

which has been peer reviewed and accepted as our advice to developers, regardless of SDK.

Reply:
That is true, although somehow I am not comfortable making users search for extra docs on how to use `ABLY_KEY`. It's a sensitive use-case from security point of view. I don't see any harm mentioning it twice. In fact, current authentication section is enough for any user to understand and setup auth. in their app without looking for extra docs at other places.
We are doing the same in ably-dotnet -> https://github.com/ably/ably-dotnet?tab=readme-ov-file#authentication

Reply:
> I am not comfortable making users search for extra docs on how to use ABLY_KEY

I understand and agree that we should be making this as easy as possible for developers, and I'm not proposing we make the user search for extra docs, I'm proposing that we link to specific sections of our official documentation which explain things regardless of SDK, rather than explaining things using bullet points in each SDK's README.

For example, rather than these two bullet points which link to the general Authentication docs and then try to explain something which is already explained in a specific section in the docs:

- It is recommended to use `ABLY_KEY` at server side. Check [official ably auth documentation](https://ably.com/docs/auth) for more info.
- `ABLY_KEY` should not be exposed at client side where it can be used for malicious purposes.

How about:

ably-go is typically used in a server-side environment where it is recommended to use an API key for authentication (for further details on why, see [Available authentication mechanisms](https://ably.com/docs/auth#mechanisms)).

and rather than:

- Token requests are issued by your servers and signed using your private API key.
- This is the preferred method of authentication as no secrets are ever shared, and the token request can be issued to trusted clients without communicating with Ably.

(followed by a code snippet)

how about:

ably-go can be used to issue tokens to clients as outlined in the [Ably Tokens documentation](https://ably.com/docs/auth/token?lang=go#tokens).

(and that section includes Go code snippets)

> We are doing the same in ably-dotnet

Same goes for those docs too then.

Reply:
Okay, let me check on this one

Reply:
updated -> https://github.com/ably/ably-go/pull/669/commits/1ca24afd1d21a3ba74ad01d5c2e52a4bb89c218c

Reply:
Okay, I have updated as per @AndyTWF review comments https://github.com/ably/ably-go/pull/669/commits/a59b9d62aa88e24785258cc1628f3bfac539465a

Reply:
Code snippets for ably-go on official docs looks incomplete :(

Reply:
I still think we should be linking to sections of our official documentation rather than repeating here what is documented there, but I don't want to block the code change being made here, so happy to approve.

Reply:
updated https://github.com/ably/ably-go/pull/669/commits/ce1c6fbb53b1e6c885069337b6ffbc5a631c4387